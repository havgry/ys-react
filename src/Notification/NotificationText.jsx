import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

const NotificationText = (props) => {
  const {
    children,
    className,
  } = props

  const wrapperProps = {
    className: classNames(
      'ys-notification__text',
      className,
    ),
  }

  return (
    <p {...wrapperProps}>
      {children}
      {/* <a href="#" class="ys-notification__inline-link"></a> */}
    </p>
  )
}

NotificationText.propTypes = {
  children: PropTypes.string.isRequired,
  className: PropTypes.string,
}

NotificationText.defaultProps = {
  className: '',
}

export default NotificationText
