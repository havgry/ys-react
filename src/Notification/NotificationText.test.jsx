import React from 'react'
import { shallow, mount as mountComponent } from 'enzyme'

import NotificationText from './NotificationText'

const customClassName = 'parrot'
const getNotificationText = (props, children = 'Notification message') => (
  <NotificationText
    {...props}
  >
    {children}
  </NotificationText>
)

const shallowRender = (props, children) => shallow(getNotificationText(props, children))
const mount = (props, children) => mountComponent(getNotificationText(props, children))

describe('<NotificationText>', () => {
  describe('Props', () => {
    it('Throws propType warning if no `children` prop is passed', () => {
      const component = () => shallowRender(null, null)
      expect(component).toThrow()
    })

    it('Has correct defaultProps', () => {
      const component = mount()

      const {
        className,
      } = component.props()

      expect(className).toBe('')
    })
  })

  describe('Rendering', () => {
    it('Outputs elements with the correct class names', () => {
      const component = shallowRender()

      expect(component.find('.ys-notification__text').exists()).toBe(true)
    })

    it('Custom `className` is being applied to the container', () => {
      const component = shallowRender({
        className: customClassName,
      })

      expect(component.find(`.ys-notification__text.${customClassName}`).exists()).toBe(true)
    })
  })

  describe('Snapshots', () => {
    it('With custom `className`', () => {
      const component = shallowRender({
        className: customClassName,
      })

      expect(component).toMatchSnapshot()
    })

    it('Without custom `className`', () => {
      const component = shallowRender()

      expect(component).toMatchSnapshot()
    })
  })
})
