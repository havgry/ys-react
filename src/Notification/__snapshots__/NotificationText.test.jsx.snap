// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationText> Snapshots With custom \`className\` 1`] = `
<p
  className="ys-notification__text parrot"
>
  Notification message
</p>
`;

exports[`<NotificationText> Snapshots Without custom \`className\` 1`] = `
<p
  className="ys-notification__text"
>
  Notification message
</p>
`;
